spec/services/media_storage/azure_adapter_spec.rb

```diff
@@ -0,0 +1,175 @@
+require 'spec_helper'
+
+RSpec.describe MediaStorage::AzureAdapter do
+  let(:storage_account_name) { 'tiny-watermelons' }
+  let(:container) { 'test' }
+  let(:opts) do
+    {
+      azure_prefix: 'test-uploads.zooniverse.org/',
+      azure_storage_account: storage_account_name,
+      azure_storage_access_key: 'fake',
+      azure_storage_container: container,
+      stub_responses: true
+    }
+  end
+  let(:adapter) { described_class.new(opts) }
+  let(:uri_regex) { /\A#{URI::DEFAULT_PARSER.make_regexp}\z/ }
+  let(:uuid_v4_regex) { /[a-f0-9]{8}\-[a-f0-9]{4}\-4[a-f0-9]{3}\-(8|9|a|b)[a-f0-9]{3}\-[a-f0-9]{12}/ }
+
+  context 'when opts are passed to the initializer' do
+    it 'uses default expiration values when no expiration values are passed' do
+      default = MediaStorage::AzureAdapter::DEFAULT_EXPIRES_IN
+      expect(adapter.instance_variable_get(:@get_expiration)).to eq(default)
+      expect(adapter.instance_variable_get(:@put_expiration)).to eq(default)
+    end
+
+    it 'defaults to current rails environment for the container name when no container is given' do
+      adapter = described_class.new(opts.except(:azure_storage_container))
+      expect(adapter.instance_variable_get(:@container)).to eq('test')
```

Comment:
use the attr_readers getter methods for these instance vars
```suggestion
      expect(adapter.container)).to eq('test')
```